test_src/sljitTest.c

```diff
@@ -11529,245 +11538,205 @@ static void test92(void)
 	executable_code code;
 	struct sljit_compiler *compiler = sljit_create_compiler(NULL, NULL);
 	struct sljit_label *label;
-	struct sljit_jump *jump;
-	sljit_sw buf[32];
-	sljit_s32 i;
+	struct sljit_jump *jump, *skip;
+	sljit_sw buf[12];
 
 	if (verbose)
 		printf("Run test92\n");
 
 	FAILED(!compiler, "cannot create compiler\n");
 
-	for (i = 0; i < 32; i++)
```

Comment:
The current testing works on big endian (e.g. s390x), I would not remove a lot of logic.

Reply:
The new one with the macros does as well, this initialization loop is no longer needed because the `\xff` bytes are added by the macros instead and the end result is clearer to read.

Reply:
If 32 tests is reduced to 12 then something is missing. I don't really understand the 1 letter long macros.

Reply:
Some "cases" where consolidated (eg, one checked the value, another the padding; I made them one single case preferably tested together with the same `FAILED`.

Others were removed because there are no longer writes (other than the required one by the LL/SC); the check to make sure the read succeeds is now done inside the test itself so there is no need for a `buf[]` for it.

An advantage of doing it inside the test is that it also verifies our assumption that reads to the "reserved area" are OK and would hopefully make it more visible if that assumption breaks with other implementations (ex: we might have a possible infloop with power)